src/main/java/core/basesyntax/service/strategy/ShopBalanceStrategy.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.db.ChangedData;
+
+public class ShopBalanceStrategy implements ShopActivityStrategy {
+    private final ChangedData changedData;
+
+    public ShopBalanceStrategy(ChangedData changedData) {
+        this.changedData = changedData;
+    }
+
+    @Override
+    public void workWithActivities(String fruit, int quantity) {
+        changedData.getChangedData().put(fruit,quantity);
+    }
+}
```

Comment:
I think it will better to create a new class ( for example Operation or FruitService) with method , which put this result to db. And you will call this new method in all type of stratages.